tests/test_categories.py

```diff
@@ -60,4 +69,38 @@ def test_invalid_path():
     with pytest.raises(Exception):
         cc_client.update_categories('file', target=invalid_categories[[0]])
     with pytest.raises(Exception):
-        cc_client.update_categories('file', target=invalid_categories[[1]])
\ No newline at end of file
+        cc_client.update_categories('file', target=invalid_categories[[1]])
+
+def test_update_account_categories_with_list():
+    """Test that should get error while updating account categories with target as a list type"""
+    cc_client = CashCtrlClient()
+    with pytest.raises(ValueError):
+        cc_client.update_categories('account', target=categories)
+
+def test_update_file_categories_with_dict():
+    """Test that should get error while updating file categories with target as a dict type"""
+    cc_client = CashCtrlClient()
+    with pytest.raises(ValueError):
+        cc_client.update_categories('file', target=account_categories)
+
+def test_account_category_update():
+    """Test that new updates categories for accounts and then restores initial state"""
+    cc_client = CashCtrlClient()
+    initial_categories = cc_client.list_categories('account')
+
+    cc_client.update_categories('account', target=account_categories)
+    remote = cc_client.list_categories('account')
+    remote = remote.drop(columns=[col for col in remote.columns if col not in ['path', 'number']])
+    target_df = pd.DataFrame(list(account_categories.items()), columns=['path', 'number'])
+    merged = pd.merge(target_df, remote, how='left', indicator=True)
+    assert (merged['_merge'] == 'both').all(), (
+        "Not all categories were updated")
```

Comment:
Why not simply compare whether dicts are equal?
```
 remote = cc_client.list_categories('account')
 remote = remote.set_index('path')['number'].to_dict()
 assert remote == account_categories, "Not all categories were updated"
```
google search for "python compare if two dicts are equal":
https://stackoverflow.com/questions/53348959/python3-determine-if-two-dictionaries-are-equal

(Also applies further below)